src/Sarif/Visitors/InsertOptionalDataVisitor.cs

```diff
@@ -126,9 +127,17 @@ internal FileData VisitDictionaryValueNullChecked(string key, FileData node)
             {
                 fileLocation.TryReconstructAbsoluteUri(_run.OriginalUriBaseIds, out Uri uri);
 
-                // TODO: we should convert node.Encoding to a .NET equivalent and pass it here
-                // https://github.com/Microsoft/sarif-sdk/issues/934
-                node = FileData.Create(uri, _dataToInsert, node.MimeType, encoding: null);
+                Encoding encoding = null;
+
+                try
```

Comment:
>try [](start = 16, length = 3)

test for non-null node.Encoding first, and don't enter this code path for that case. will prevent us raising/catching exceptions unnecessarily.